### PR TITLE
Refactor: Decouple Social's ToggleSection component

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change decouples the `ToggleSection` component, so it's not tied to the logic behind it.

## Proposed changes:

* Decoupled the `ToggleSection` component, now it has props and handlers.
* Added some type definitions to support our motion to use more TypeScript with our components. p1678866772214549-slack-C02JJ910CNL
* Created a new `SocialModuleToggle` component that utilizes the decoupled `ToggleSection` component
* Removed the `ModuleToggle` as after decoupling it was only some styling to the `ToggleControl` component

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1204116019981263-as-1204237713388861

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Go to the Social Admin page, and see if the Publicize toggle is still there and still works.
* If it's off, you should not see the sharing section in the editor.
* Changing the state should also be preserved and loaded upon page refresh

### CODE REVIEW

File: projects/plugins/social/declaration.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+declare module '*.module.css' {
+	const classes: { [ key: string ]: string };
+	export default classes;
+}
+
+declare module '*.module.scss' {
+	const classes: { [ key: string ]: string };
+	export default classes;
+}
```
Comment (danielpost) [inline projects/plugins/social/declaration.d.ts]: What does this file do? Just curious!

File: projects/plugins/social/src/js/components/social-module-toggle/index.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import ToggleSection from '../toggle-section';
+import { SocialStoreSelectors } from '../types/types';
+import styles from './styles.module.scss';
+
+/**
+ * This component is used in the Social admin panel to toggle the Publicize module.
+ *
+ * @returns {React.Component} The SocialModuleToggle component.
+ */
+export default function SocialModuleToggle(): JSX.Element {
```
Comment (danielpost) [inline projects/plugins/social/src/js/components/social-module-toggle/index.tsx]: `React.Component` and `JSX.Element` aren't exactly the same thing, right? Should these both be the same or does it not matter?

File: projects/plugins/social/src/js/components/toggle-section/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {ToggleSectionProps} props - Component props.
+ * @returns {React.ReactNode} - ToggleSection react component.
+ */
+export default function ToggleSection( {
+	title,
+	onChange,
+	checked,
+	disabled,
+	children,
+}: ToggleSectionProps ): JSX.Element {
```
Comment (danielpost) [inline projects/plugins/social/src/js/components/toggle-section/index.tsx]: Similar here, it says `React.ReactNode` and `JSX.Element`. Should these be consistent?

File: projects/plugins/social/src/js/components/types/types.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+type JetpackSettingsSelectors = {
+	getJetpackSettings: () => {
+		publicize_active: boolean;
+		show_pricing_page: boolean;
+		showNudge: boolean;
```
Comment (danielpost) [inline projects/plugins/social/src/js/components/types/types.ts:5]: I know it wasn't introduced in this PR (and should be fixed in a separate PR) but we should really be consistent here with casing 😬

**Review by danielpost — DISMISSED** [review]
This looks good to me and works well! I can't comment too much on the TypeScript changes but they look good—I added a couple thoughts. Feel free to address or ignore them.

File: projects/plugins/social/declaration.d.ts (hunk 1 above)
Comment (gmjuhasz) [reply projects/plugins/social/declaration.d.ts]: I had some errors, because with TypeScript you can only import valid modules, and without this the usual scss import we do `import styles from './styles.module.scss';` would show an error, because it cannot recognise syntax there. This defines that the scss files are generally arrays that have named objects with string key-value pairs. I found this type of solution in other parts of the repo too (like in the js-packages folder).

File: projects/plugins/social/src/js/components/social-module-toggle/index.tsx (hunk 2 above)
Comment (gmjuhasz) [reply projects/plugins/social/src/js/components/social-module-toggle/index.tsx]: Yeah, good catch. I am not sure either, but I saw that JSX.Element is more widely used, so I think I will change the other one to that too.

File: projects/plugins/social/declaration.d.ts (hunk 1 above)
Comment (gmjuhasz) [reply projects/plugins/social/declaration.d.ts]: Update: I ended up using the bas tsconfig instead which already does this

**Review by danielpost — APPROVED** [review]
